app/bmc/src/main.c

```diff
@@ -23,10 +23,13 @@
 #include <tenstorrent/tt_smbus.h>
 #include <tenstorrent/bh_chip.h>
 #include <tenstorrent/bh_arc.h>
+#include <tenstorrent/bmc_main_queue.h>
```

Comment:
Can we not include this header please?

It would be cleaner to just call the functions directly.

There are also other ways to wait for an event. E.g. k_poll, k_event, etc.

Those allow you to wake on multiple kernel objects or multiple different types of events.